### ISSUE TITLE
Improve documentation for subscriptions

### ISSUE THREAD

**Issue by b1zzu** [aerogear/offix#252]
## Feature Request

**Is your feature request related to a problem? Please describe.**
I have problems in understanding how to use the offix subscription helper methods, and I think that the documentation around subscriptions is outdated, for example, the `wsUrl` options, seems to not exist anymore, but is still documented.

**Describe the solution you'd like**
I would like to see some full working examples in the documentation.

**Describe alternatives you've considered**
Reading the code

**Comment by darahayes** [comment]
Hey @b1zzu, we used to have the `WebsocketLink` being set up inside the `OfflineClient` but we got many complaints because some people didn't need it and others wanted to configure it themselves. We no longer set it up at all. Instead we provide a [`terminatingLink` option](https://offix.dev/docs/client-configuration). You can use this to achieve what you need. The code for it would look something like this:

```
import { split } from 'apollo-link';
import { HttpLink } from 'apollo-link-http';
import { WebSocketLink } from 'apollo-link-ws';
import { getMainDefinition } from 'apollo-utilities';
import { OfflineClient } from 'offix-client';

// Create an http link:
const httpLink = new HttpLink({
  uri: 'http://localhost:3000/graphql'
});

// Create a WebSocket link:
const wsLink = new WebSocketLink({
  uri: `ws://localhost:5000/`,
  options: {
    reconnect: true
  }
});

// create the composite link
const terminatingLink = split(
  // split based on operation type
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === 'OperationDefinition' &&
      definition.operation === 'subscription'
    );
  },
  wsLink,
  httpLink,
);

const client = new OfflineClient({ terminatingLink })
```

**Comment by b1zzu** [comment]
Thanks, @darahayes! but what about the `createSubscriptionOptions` helper? Are u going to remove also that one?

**Comment by wtrocki** [comment]
Subscriptions need to be documented. 
Mostly by mentioning offline patterns and helpers we provide.

**Comment by darahayes** [comment]
@b1zzu TL;DR - we do not set up subscriptions anymore but `createSubscriptionOptions` is being kept for the time being. However it will need to be reviewed now that Apollo 3.0 is out/

To add context to @wtrocki's reply. `createSubscriptionOptions` is a helper function that returns some options containing special cache `update` functions that help keep things in the Apollo cache properly maintained while subscriptions are being used in the wider context of an 'offline application.'

So while we do not do set up the websocket/subscriptions capabilities anymore. (There are too many options and some people don't need it at all), The `createSubscriptionOptions` method is generally still useful and will not be going away.

That being said, since Apollo 3 came out recently, the way the cache can be managed by the end user is a little different now so we will need to review that very soon and it's possible that it could be deprecated in the future. We need to do more investigation there to understand it fully.

**Comment by b1zzu** [comment]
Thanks for the explanation, but the question now is if it would make sense to update the documentation with the information you give me? We could also add @darahayes  example in the documentation.

Or should we postpone this work until we don't have a clear idea of how Apollo 3 will look like?

Wdyt?

**Comment by wtrocki** [comment]
@darahayes Let's add this to roadmap after integration with Apollo 3.0

**Comment by kingsleyzissou** [comment]
We are closing this issue as we are currently working on a refactor of the subscription engine & the underlying support. We will update the docs once this is complete.
